### DISCUSSION TITLE
error: Uncaught (in promise) AssertionError

### DISCUSSION THREAD

**Discussion by goodmanworld** [denoland/deno#12377] [Q&A] [answered]
I am a new Deno learner, I tried a tutorial code that uses MongoDB but I got the following error:

```
error: Uncaught (in promise) AssertionError
    throw new AssertionError(msg);
          ^
    at assert (https://deno.land/std@0.107.0/testing/asserts.ts:224:11)
    at MongoClient.database (https://deno.land/x/mongo@v0.27.0/src/client.ts:67:5)
    at file:///C:/Users/m/Desktop/Uproject/GuidApp/deno-survey/mongo.ts:6:19
```

As the project has many files, Let me know if is it needed to add some parts of project codes here?
But the whole project code can be find here: https://github.com/thecodeholic/deno-survey

**Comment by kitsonk** [comment]
Please stop opening issues for general usage questions. Either join the Discord forum or limit your questions to the Q&A section of the discussion forum.

**Comment by erfanium** [comment, marked as answer]
You should connect the `MongoClient` before using its methods:

```ts
const client = new MongoClient();

await client.connect("mongodb://localhost:27017");
// now client is connected, you can use it

const db = client.database("test");
const users = db.collection<UserSchema>("users");
```

Please open a discussion or issue in https://github.com/denodrivers/deno_mongo if you still have problems
